src/state/sagas/datasets/UpdateDatasetSecurity/UpdateDatasetSecurity.js

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT licence.
+import { t } from 'i18next';
+import { takeEvery, select, call, put } from 'redux-saga/effects';
+import DatasetService from '../../../../services/dataset/DatasetService';
+import RunnerService from '../../../../services/runner/RunnerService';
+import { DATASET_ACTIONS_KEY } from '../../../commons/DatasetConstants';
+import { dispatchSetApplicationErrorMessage } from '../../../dispatchers/app/ApplicationDispatcher';
+
+const getOrganizationId = (state) => state.organization.current.data.id;
+const getWorkspaceId = (state) => state.workspace.current.data.id;
+const getDatasetListData = (state) => state.dataset.list.data;
+
+export function* updateDatasetSecurity(action) {
+  const datasetId = action.datasetId;
+  try {
+    const organizationId = yield select(getOrganizationId);
+    const workspaceId = yield select(getWorkspaceId);
+    const datasetListData = yield select(getDatasetListData);
+    const dataset = datasetListData.find((dataset) => dataset.id === datasetId);
+    const oldDatasetSecurity = dataset?.security;
+    const newDatasetSecurity = action.datasetSecurity;
+    const currentUserPermissions = dataset.security.currentUserPermissions;
+
+    yield call(DatasetService.updateSecurity, organizationId, datasetId, oldDatasetSecurity, newDatasetSecurity);
+
+    const datasetUpdated = { ...dataset, security: { ...newDatasetSecurity, currentUserPermissions } };
```

Comment:
I can be mistaken but here, we are storing only newSecurity, not the updated one ? :thinking:

Reply:
In the Api function, the updated security parameter is called _newSecurity_ in opposition as _oldSecurity_. So I tryed to keep a name as close as possible to the parameter name.